reVX/least_cost_xmission/aoswt_utilities.py

```diff
@@ -0,0 +1,604 @@
+                                     '.profile()!')
+        self._os_shape = (self.profile()['height'],
+                          self.profile()['width'])
+
+        self._os_barriers = None  # (uint8) off-shore barrier raster
+        self._os_friction = None  # (float32) off-shore friction raster
+        self._land_mask = None  # (bool) land mask raster, true indicates land
+
+    def create_land_mask(self, mask_shp_f, save_tiff=False):
+        """
+        Create the land mask layer from a vector file
+
+        Parameters
+        ----------
+        mask_shp_f : str
+            Full path to mask gpgk or shp file
+        save_tiff : bool
+            Save mask as tiff if true
+        """
+        ldf = gpd.read_file(mask_shp_f)
+        print(f'Rasterizing {mask_shp_f}')
+        l_geom = list(ldf.geometry)
+        l_rast = features.rasterize(l_geom, out_shape=self._os_shape, fill=0,
+                                    out=None,
+                                    transform=self.profile()['transform'],
+                                    all_touched=False, default_value=1,
+                                    dtype=None)
+
+        if save_tiff:
+            print(f'Saving land mask to {self.LAND_MASK_FNAME}')
+            self._save_tiff(l_rast, self.LAND_MASK_FNAME)
+
+        self._land_mask = l_rast == 1
+        print('Rasterizing complete')
+
+    def load_land_mask(self, mask_f=None):
+        """
+        Load the land mask layer from a tiff. This does not need to be called
+        if self.create_land_mask() was run previously.
+
+        Parameters
+        ----------
+        mask_f : str
+            Full path to mask tiff
+        """
+        if mask_f is None:
+            mask_f = self.LAND_MASK_FNAME
+        with rio.open(mask_f) as ras:
+            l_rast = ras.read(1)
+
+        assert l_rast.max() == 1
+        assert l_rast.shape == self._os_shape
+
+        self._land_mask = l_rast == 1
+        print(f'Successfully loaded land mask from {mask_f}')
+
+    def build_off_shore_friction(self, friction_files, slope_file=None,
+                                 save_tiff=None):
+        """
+        Combine off-shore friction layers.
+
+        friction_files : list of tuples
+            Friction layers to combine and raster value to friction dict.
+            Tuples are in the format:
+
+                ({ras_val1: fric_val1, ras_val2: fric_val2, ...}, 'fname.tif')
+
+            where ras_valX is the value in the raster, and fric_valX is the
+            desired friction values. Any values in the raster that are not
+            specified in the dict are assumed to have no friction. 'fname.tif'
+            is the file name of the raster.
+        slope_file : str, optional
+            Path to slope friction tiff
+        save_tiff : bool, optional
+            Save composite friction to tiff if true
+        """
+        print('Loading friction layers')
```

Comment:
Please convert all print statements to logger.info() statements

Reply:
Done